server/src/main/java/io/crate/execution/ddl/tables/AddColumnRequest.java

```diff
@@ -72,10 +71,11 @@ public AddColumnRequest(StreamInput in) throws IOException {
         }
 
         int numColumns = in.readVInt();
-        colsToAdd = new ArrayList<>(numColumns);
+        var cols = new ArrayList<Reference>(numColumns);
         for (int i = 0; i < numColumns; i++) {
-            colsToAdd.add(Reference.fromStream(in));
+            cols.add(Reference.fromStream(in));
         }
+        this.colsToAdd = List.copyOf(cols);
```

Comment:
Why add the copy here? It already creates a new `ArrayList` that it owns.

Reply:
was addressing comment to make list immutable (once we built it fully). https://github.com/crate/crate/pull/13383#discussion_r1053077643

Reply:
We usually don't make defensive copies in cases like this but instead follow the rule to only mutate lists a component owns.

It also wouldn't be consistent, `checkConstraints` and the primaryKeyIndices would still be mutable.

Reply:
+1 to inconsistent.

> only mutate lists a component owns.
does it it mean we might want to receive an immutable copy in normal constructor (when it's created outside)?

what about `references()` where we expose created in the component (currently read only usages) - do we say that we don't need a copy if other accuser are not mutating?